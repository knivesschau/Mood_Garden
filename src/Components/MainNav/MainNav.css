/* mobile style, starting at 320px*/
.MainNav_NoLogin {
   display: flex;
   flex-direction: row;
   justify-content: space-evenly;
   margin-left: auto;
   margin-right: auto;
   font-size: 1.25em;
   font-weight: bold;
   text-transform: uppercase;
}

.MainNav_SignedIn {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    font-size: 1.25em;
    font-weight: bold;
    text-transform: uppercase;
}

#register-link, #login-link, #plant-rose, #home-page, #logout-link {
    transition-property: background;
    transition-duration: .2s;
    transition-timing-function: ease-in; 
}

#register-link:hover, #login-link:hover, #plant-rose:hover, #home-page:hover, #logout-link:hover {
    cursor: pointer;
    background-color: #FFADBB;
    color: #FFFFFF;
}

#register-link:active, #login-link:active, #plant-rose:active, #home-page:active, #logout-link:active {
    background-color: #8F0000;
}

@media only screen and (min-device-width: 414px) {
    .MainNav_NoLogin, .MainNav_SignedIn {
        font-size: 1.50em;
    }
}

@media only screen and (min-device-width: 768px) {
    .MainNav_NoLogin {
        margin-left: 3em;
        margin-right: 3em;
    }

    .MainNav_SignedIn {
        margin-left: 5em;
        margin-right: 5em;
    }
}

@media only screen and (min-device-width: 1000px) {
    .MainNav_NoLogin, .MainNav_SignedIn {
        font-size: 2em;
    }

    .MainNav_NoLogin, .MainNav_SignedIn {
        margin-left: 6em;
        margin-right: 6em;
    }

}

@media only screen and (min-device-width: 1200px) {
    .MainNav_NoLogin {
        margin-left: 7.5em;
        margin-right: 7.5em;
    }

    .MainNav_SignedIn {
        margin-left: 10em;
        margin-right: 10em;
    }
}

@media only screen and (min-device-width: 1400px) {
    .MainNav_NoLogin, .MainNav_SignedIn {
        font-size: 2.15em;
    }
    
    .MainNav_NoLogin {
        margin-left: 9.5em;
        margin-right: 9.5em;
    }
} 