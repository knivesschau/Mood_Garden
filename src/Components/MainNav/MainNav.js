import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import TokenService from '../../services/token-services';
import './MainNav.css';

export default class MainNav extends Component {
    handleLogoutClick = () => {
        TokenService.clearAuthToken();
        window.location='/';
    }

    renderUserNav() {
        return (
            <div className="MainNav_SignedIn">
                <Link to='/plant-rose' id="plant-rose">
                    Plant 
                </Link>

                <FontAwesomeIcon icon="leaf" color="green"/>
                
                <Link to='/' id="home-page">
                    About
                </Link>

                <FontAwesomeIcon icon="leaf" color="green"/>
                
                <Link to='/' id="logout-link" onClick={this.handleLogoutClick}>
                    Logout 
                </Link>

            </div>
        )
    }

    renderStartingNav() {
        return (
            <div className="MainNav_NoLogin">
                <Link to='/register' id="register-link">
                     Sign Up
                </Link>

                <FontAwesomeIcon icon="seedling" color="green"/>

                <Link to='/login' id="login-link">
                     Log In
                </Link>
            </div>
        )
    }
    
    render() {
        return (
            <nav role="navigation" className="Main_Nav">

                {TokenService.hasAuthToken()
                    ? this.renderUserNav()
                    : this.renderStartingNav()}
                    
            </nav>
        )
    }
}