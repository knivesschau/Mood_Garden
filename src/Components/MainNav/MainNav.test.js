import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import MainNav from './MainNav';

describe ('MainNav Component', () => {
    it ('renders the complete MainNav Component', () => {
        const wrapper = shallow(<MainNav/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})


