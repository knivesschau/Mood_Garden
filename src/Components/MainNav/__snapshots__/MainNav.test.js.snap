// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav Component renders the complete MainNav Component 1`] = `
<nav
  className="Main_Nav"
  role="navigation"
>
  <div
    className="MainNav_NoLogin"
  >
    <Link
      id="register-link"
      to="/register"
    >
      Sign Up
    </Link>
    <FontAwesomeIcon
      border={false}
      className=""
      color="green"
      fixedWidth={false}
      flip={null}
      icon="seedling"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <Link
      id="login-link"
      to="/login"
    >
      Log In
    </Link>
  </div>
</nav>
`;
